File: typescript-connector/src/models/ProductionFlow.ts
// Class from DFC Business Ontology: #ProductionFlow
import { SemanticObject } from "../core/SemanticObject.js";
import type { QuantitativeValue } from "./QuantitativeValue.js";

export interface ProductionFlowParams {
  quantity?: number;
  outputOf?: string;
  produces?: string[];
  date?: string;
  description?: string;
  name?: string;
  characteristicOf?: string;
  hasDimension?: string;
  hasQuantity?: QuantitativeValue;
}
export class ProductionFlow extends SemanticObject {
  static get SEMANTIC_TYPE(): string {
    return "dfc-b:ProductionFlow";
  }

  quantity?: number;
  outputOf?: string;
  produces?: string[];
  date?: string;
  description?: string;
  name?: string;
  characteristicOf?: string;
  hasDimension?: string;
  hasQuantity?: QuantitativeValue;

  constructor(
    semanticId: string,
    params?: ProductionFlowParams,
  ) {
    super(semanticId);
    this.quantity = params?.quantity;
    this.outputOf = params?.outputOf;
    this.produces = params?.produces;
    this.date = params?.date;
    this.description = params?.description;
    this.name = params?.name;
    this.characteristicOf = params?.characteristicOf;
    this.hasDimension = params?.hasDimension;
    this.hasQuantity = params?.hasQuantity;
    this.semanticType = ProductionFlow.SEMANTIC_TYPE;
    this.registerSemanticProperty("dfc-b:quantity", () => this.quantity);
    this.registerSemanticProperty("dfc-b:outputOf", () => this.outputOf);
    this.registerSemanticProperty("dfc-b:produces", () => this.produces);
    this.registerSemanticProperty("dfc-b:date", () => this.date);
    this.registerSemanticProperty("dfc-b:description", () => this.description);
    this.registerSemanticProperty("dfc-b:name", () => this.name);
    this.registerSemanticProperty("dfc-b:characteristicOf", () => this.characteristicOf);
    this.registerSemanticProperty("dfc-b:hasDimension", () => this.hasDimension);
    this.registerSemanticProperty("dfc-b:hasQuantity", () => this.hasQuantity);
  }
  static {
    SemanticObject.typeRegistry.set(ProductionFlow.SEMANTIC_TYPE, ProductionFlow);
  }
}

File: ruby-gem/lib/models/production_flow.rb
# frozen_string_literal: true

# Class from DFC Business Ontology: #ProductionFlow
require_relative '../core/semantic_object'


module DfcLinkmlConnector
  module Models
    class ProductionFlow < Core::SemanticObject
      SEMANTIC_TYPE = "dfc-b:ProductionFlow".freeze

      # @return [Float]
      attr_accessor :quantity

      # @return [String]
      attr_accessor :output_of

      # @return [String]
      attr_accessor :produces

      # @return [String]
      attr_accessor :date

      # @return [String]
      attr_accessor :description

      # @return [String]
      attr_accessor :name

      # @return [String]
      attr_accessor :characteristic_of

      # @return [String]
      attr_accessor :dimension

      # @param semanticId [String]
      # @param quantity: nil, outputOf: nil, produces: nil, date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil
      def initialize(semanticId, quantity: nil, outputOf: nil, produces: nil, date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil)
        super(semanticId)
        @quantity = quantity
        @output_of = outputOf
        @produces = produces
        @date = date
        @description = description
        @name = name
        @characteristic_of = characteristicOf
        @dimension = dimension
        self.semanticType = "dfc-b:ProductionFlow"
        registerSemanticProperty("dfc-b:quantity", &method("quantity")).valueSetter = method("quantity=")
        registerSemanticProperty("dfc-b:outputOf", &method("output_of")).valueSetter = method("output_of=")
        registerSemanticProperty("dfc-b:produces", &method("produces")).valueSetter = method("produces=")
        registerSemanticProperty("dfc-b:date", &method("date")).valueSetter = method("date=")
        registerSemanticProperty("dfc-b:description", &method("description")).valueSetter = method("description=")
        registerSemanticProperty("dfc-b:name", &method("name")).valueSetter = method("name=")
        registerSemanticProperty("dfc-b:characteristicOf", &method("characteristic_of")).valueSetter = method("characteristic_of=")
        registerSemanticProperty("dfc-b:hasDimension", &method("dimension")).valueSetter = method("dimension=")
      end

      Core::SemanticObject.type_registry[SEMANTIC_TYPE] = self
    end
  end
end
